src/app/technical-record-search/test-result.service.ts

```diff
@@ -15,21 +19,87 @@ export class TestResultService {
   private _apiServer = this._appConfig.settings.apiServer;
   private readonly routes;
 
-  constructor(private httpClient: HttpClient,
+  constructor(
+    private httpClient: HttpClient,
     private _store: Store<IAppState>,
-    private _appConfig: AppConfig) {
+    private _appConfig: AppConfig
+  ) {
     this.routes = {
-      testResults: (searchIdentifier: string) => `${this._apiServer.APITestResultServerUri}/test-results/${searchIdentifier}`,
+      testResults: (searchIdentifier: string) =>
+        `${this._apiServer.APITestResultServerUri}/test-results/${searchIdentifier}`,
+      preparers: () => `${this._apiServer.APIPreparersServerUri}/preparers`,
+      testStation: () => `${this._apiServer.APITestStationsServerUri}/test-stations`,
+      updateTestResults: (systemNumber: string) =>
+        `${this._apiServer.APITestResultServerUri}/test-results/${systemNumber}`,
+      testResultById: (systemNumber: string, testResultId: string) =>
+        `${this._apiServer.APITestResultServerUri}/test-results/${systemNumber}?testResultId=${testResultId}&version=all`
     };
   }
 
-  getTestResults(searchIdentifier: string): Observable<any> {
+  getTestResults(searchIdentifier: string): Observable<TestResultModel> {
     this._store.dispatch(new LoadingTrue());
     const headers = new HttpHeaders().set('Content-Type', 'application/json; charset=utf-8');
-    return this.httpClient.get<any[]>(this.routes.testResults(searchIdentifier), { headers }).pipe(
-      delayedRetry(),
-      shareReplay(),
-      finalize(() => this._store.dispatch(new LoadingFalse()))
-    );
+    return this.httpClient
+      .get<TestResultModel[]>(this.routes.testResults(searchIdentifier), { headers })
+      .pipe(
+        delayedRetry(),
```

Comment:
This needs to be covered in the unit test perhaps? Like spies and that they are being called. I am not sure what is the most appropriate approach to mock that. Maybe we need to mock the entire module?
Thoughts @alleypa ?

Reply:
As agreed with the team, the code in place was prior to ADR and a technical task has been added to trim the effects where possible
Reference: https://jira.dvsacloud.uk/browse/CVSB-15100